tests/test_daemon.cpp

```diff
@@ -988,4 +992,38 @@ TEST_F(Daemon, reads_mac_addresses_from_json)
     check_interfaces_in_json(filename, mac_addr, extra_interfaces);
 }
 
+TEST_F(Daemon, launches_with_valid_network_interface)
+{
+    mpt::MockVirtualMachineFactory* mock_factory = use_a_mock_vm_factory();
+
+    mp::Daemon daemon{config_builder.build()};
+
+    EXPECT_CALL(*mock_factory, list_networks());
+
+    ASSERT_NO_THROW(send_command({"launch", "--network", "eth0"}));
+}
+
+TEST_F(Daemon, refuses_launch_with_invalid_network_interface)
+{
+    mpt::MockVirtualMachineFactory* mock_factory = use_a_mock_vm_factory();
+
+    mp::Daemon daemon{config_builder.build()};
+
+    EXPECT_CALL(*mock_factory, list_networks());
+
+    std::stringstream err_stream;
+    send_command({"launch", "--network", "eth2"}, std::cout, err_stream);
+    EXPECT_THAT(err_stream.str(), HasSubstr("Invalid network options supplied"));
```

Comment:
Of course... thanks for reminding my sleepy self of this today, I had forgotten to suggest it in this other PR.